src/test/pybind/test_rbd.py

```diff
@@ -1742,6 +1742,11 @@ def test_clone_format(self):
             eq(RBD_OPERATION_FEATURE_CLONE_CHILD, clone2.op_features())
         self.rbd.remove(ioctx, clone_name2)
 
+    def test_unsupport_clone_format(self):
+        clone_name2 = get_temp_image_name()
+        assert_raises(InvalidArgument, self.rbd.clone, ioctx, image_name,
+                       'snap1', ioctx, clone_name2, features, clone_format=11)
```

Comment:
I'd add test `0` and `3` instead.

But Python tests are too weak for this -- for example there is no way to set `RBD_IMAGE_OPTION_FORMAT` for `clone()`, so the fact that validation is currently missing there can't really be tested.  Let's also add C and/or C++ tests, passing in `opts` with `RBD_IMAGE_OPTION_FORMAT` set to:
- 0 and 3 for e.g. `rbd_create4()`, `rbd_copy3()` and `rbd_copy4()`
- 0, 1 and 3 for e.g.  `rbd_clone3()`, `rbd_clone4()`, `rbd_migration_prepare()`, `rbd_migration_prepare_import()`, `rbd_deep_copy()`

and `opts` with `RBD_IMAGE_OPTION_CLONE_FORMAT` set to 0 and 3 for e.g. `rbd_clone3()`, `rbd_clone4()` and any value for the rest of the above APIs that shouldn't accept `RBD_IMAGE_OPTION_CLONE_FORMAT`.

All of these would be negative tests, expecting an error.